simulated_data_sources.go

```diff
@@ -38,19 +38,27 @@ func (ts *TriangleSource) Configure(config *TriangleSourceConfig) error {
 	}
 	ts.runMutex.Lock()
 	defer ts.runMutex.Unlock()
+	if config.Min > config.Max {
+		return fmt.Errorf("have config.Min=%v > config.Max=%v, want Min<Max", config.Min, config.Max)
+	}
+	nrise := config.Max - config.Min
+	fmt.Println(nrise)
+	if nrise > 0 {
+		ts.cycleLen = 2 * int(nrise)
+		ts.onecycle = make([]RawType, ts.cycleLen)
+		var i RawType
+		for i = 0; i < nrise; i++ {
+			ts.onecycle[i] = config.Min + i
+			ts.onecycle[int(i)+int(nrise)] = config.Max - i
+		}
+	} else if nrise == 0 {
+		ts.cycleLen = 1
```

Comment:
Why not make this bigger, say, 10000?

Reply:
thats probably a good idea, i just followed the natural extension of the previous code. Maybe there should be a minimum cycle length?